include/numerics/type_n_tensor.h

```diff
@@ -205,9 +217,37 @@ class TypeNTensor
     t.print(os);
     return os;
   }
+
+  /**
+   * Add a scaled type N tensor to this type N tensor without creating a temporary.
+   */
+  template<typename T2>
+  void add_scaled (const TypeNTensor<N, T2> &, const T &);
+
+  /**
+   * The coordinates of the \p TypeNTensor
+   */
+  std::vector<T> _coords;
+
+private:
+  static int int_pow(int b, int e)
```

Comment:
There's probably some constexpr tricks that we ought to do to make sure that the compiler knows not to actually evaluate this every time it's invoked.  But that can wait until someone actually uses it.